File: CS/ViewModels/Base/UserInfoViewModel.cs
namespace DXHtmlMessengerSample.ViewModels {
    using System.ComponentModel;
    using System.Drawing;
    using DevExpress.DevAV.Chat.Model;
    using DevExpress.Mvvm;
    using DevExpress.Mvvm.POCO;

    [EditorBrowsable(EditorBrowsableState.Never)]
    public abstract class UserInfoViewModel : IDocumentContent, ISupportParameter {
        UserInfo userInfo;
        protected UserInfoViewModel(UserInfo userInfo) {
            this.userInfo = userInfo;
        }
        object ISupportParameter.Parameter {
            get { return userInfo; }
            set {
                var info = value as UserInfo;
                if(userInfo == info) return;
                userInfo = info;
                this.RaisePropertiesChanged();
            }
        }
        public long Id {
            get { return userInfo.Id; }
        }
        public string Name {
            get { return userInfo.Name; }
        }
        public Image Photo {
            get { return userInfo.Photo; }
        }
        public string MobilePhone {
            get { return userInfo.MobilePhone; }
        }
        public string Email {
            get { return userInfo.Email; }
        }
        #region IDocumentContent
        protected void CloseDocument() {
            var owner = ((IDocumentContent)this).DocumentOwner;
            if(owner != null) owner.Close(this);
        }
        object IDocumentContent.Title { get { return string.Empty; } }
        IDocumentOwner IDocumentContent.DocumentOwner { get; set; }
        void IDocumentContent.OnClose(CancelEventArgs e) { }
        void IDocumentContent.OnDestroy() { }
        #endregion IDocumentContent
        protected internal abstract string ServiceKey { get; }
    }
}

File: VB/ViewModels/Base/UserInfoViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm
Imports DevExpress.DevAV.Chat.Model
Imports System.Drawing
Imports System.ComponentModel

Namespace DXHtmlMessengerSample.ViewModels

    <EditorBrowsable(EditorBrowsableState.Never)>
    Public MustInherit Class UserInfoViewModel
        Implements IDocumentContent, ISupportParameter

        Private userInfo As UserInfo
        Protected Sub New(ByVal userInfo As UserInfo)
            Me.userInfo = userInfo
        End Sub
        Private Property ISupportParameter_Parameter() As Object Implements ISupportParameter.Parameter
            Get
                Return userInfo
            End Get
            Set(ByVal value As Object)
                Dim info = TryCast(value, UserInfo)
                If userInfo Is info Then
                    Return
                End If
                userInfo = info
                Me.RaisePropertiesChanged()
            End Set
        End Property
        Public ReadOnly Property Id() As Long
            Get
                Return userInfo.Id
            End Get
        End Property
        Public ReadOnly Property Name() As String
            Get
                Return userInfo.Name
            End Get
        End Property
        Public ReadOnly Property Photo() As Image
            Get
                Return userInfo.Photo
            End Get
        End Property
        Public ReadOnly Property MobilePhone() As String
            Get
                Return userInfo.MobilePhone
            End Get
        End Property
        Public ReadOnly Property Email() As String
            Get
                Return userInfo.Email
            End Get
        End Property
        #Region "IDocumentContent"
        Protected Sub CloseDocument()
            Dim owner = DirectCast(Me, IDocumentContent).DocumentOwner
            If owner IsNot Nothing Then
                owner.Close(Me)
            End If
        End Sub
        Private ReadOnly Property IDocumentContent_Title() As Object Implements IDocumentContent.Title
            Get
                Return String.Empty
            End Get
        End Property
        Private Property IDocumentContent_DocumentOwner() As IDocumentOwner Implements IDocumentContent.DocumentOwner
        Sub IDocumentContent_OnClose(ByVal e As CancelEventArgs) Implements IDocumentContent.OnClose
        End Sub
        Sub IDocumentContent_OnDestroy() Implements IDocumentContent.OnDestroy
        End Sub
        #End Region ' IDocumentContent
        Protected Friend MustOverride ReadOnly Property ServiceKey() As String
    End Class
End Namespace